Tasks/VSMobileCenterUpload/utils.ts

```diff
@@ -119,16 +139,66 @@ export function resolveSinglePath(pattern: string, continueOnError: boolean): st
             }
         }
 
-        if (matches.length != 1) {
-            if (continueOnError) {
-                tl.warning(tl.loc("FoundMultipleFiles", pattern));
-            } else {
-                throw new Error(tl.loc("FoundMultipleFiles", pattern));
-            }
+        return removeDuplicates(matches.map(v => packParentFolder ? path.dirname(v) : v));
+    }
+
+    return null;
+}
+
+export function removeDuplicates(list: string[]): string[] {
+    let unique: string[] = [];
+
+    list.forEach(s => {
+        if (!unique.find(t => t === s)) {
+            unique.push(s);
         }
+    });
 
-        return matches[0];
+    return unique;
+}
+
+export function findCommonParent(list: string[]): string {
+    function cutTail(list: string[], n: number) {
+        while (n-- > 0) {
+            list.pop();
+        }
     }
 
-    return null;
-}
\ No newline at end of file
+    if (!list) {
+        return null;
+    }
+
+    let commonParent: string[] = [];
+
+    list.forEach((path, idx) => {
+        if (idx === 0) {
+            // Take the first path as the common parent candidate
+            commonParent = path.split("/");
+            return;
+        }
+
+        if (commonParent.length === 0) { 
+            // We've already detected that the paths do not have a common parent.
+            // No sense to check the rest of paths.
+            return; 
+        }
+
+        let pathSegmants: string[] = path.split("/");
+
+        // If the current path contains less segments than the common path calculated so far,
+        // the trailing segmants in the latter cannot be a part of the resulting common path.
+        cutTail(commonParent, commonParent.length - pathSegmants.length);
+
+        for (let i = 0; i < pathSegmants.length; i++) {
+            if (pathSegmants[i] !== commonParent[i]) {
+                // Segments i, i+1, etc. cannot be a part of the resulting common path.
+                cutTail(commonParent, commonParent.length - i);
+                break;
+            }
+        }
+    })
+
+    return commonParent.join("/");
+}
```

Comment:
It looks like we do not need that. Paths processed in this method always have forward slashes (even in Windows).